src/test/java/tests/FilterParserTests.java

```diff
@@ -52,69 +57,77 @@ public void quotes() {
     }
 
     @Test(expected = ParseException.class)
-    public void emptyQuotes() {
-        Parser.parse("body: \"\""); 
+    public void parse_emptyQuotes_exceptionThrown() {
+        Parser.parse("body: \"\"");
     }
 
     @Test(expected = ParseException.class)
-    public void multiplesQuotes() {
-        Parser.parse("body: \"\"\""); 
+    public void parse_unbalancedQuotes_exceptionThrown() {
+        Parser.parse("body: \"\"\"");
     }
 
+    @Test(expected = ParseException.class)
+    public void parse_quotesInPrefixPosition_exceptionThrown() {
+        Parser.parse("body: test\"will not be parsed\"\"");
+    }
 
     @Test(expected = ParseException.class)
-    public void prefixQuotes() {
-        Parser.parse("body: test\"will not be parsed\"\""); 
+    public void parse_unrecognisedTokenAsQualifierContent_exceptionThrown() {
+        Parser.parse("a:.");
     }
 
-    @Test
-    public void unexpectedEOFs() {
-        try {
-            Parser.parse("a:.");
-            fail("Inputs which end unexpectedly should throw a parse exception");
-        } catch (ParseException ignored) {}
-        try {
-            Parser.parse("a:");
-            fail("Inputs which end unexpectedly should throw a parse exception");
-        } catch (ParseException ignored) {}
-        try {
-            Parser.parse("~");
-            fail("Inputs which end unexpectedly should throw a parse exception");
-        } catch (ParseException ignored) {}
-        try {
-            Parser.parse("a(b) ||");
-            fail("Inputs which end unexpectedly should throw a parse exception");
-        } catch (ParseException ignored) {}
+    @Test(expected = ParseException.class)
+    public void parse_noQualifierContent_exceptionThrown() {
+        Parser.parse("a:");
     }
 
-    @Test
-    public void emptyQualifiers() {
+    @Test(expected = ParseException.class)
+    public void parse_standalonePrefixOperator_exceptionThrown() {
+        Parser.parse("~");
+    }
+
+    @Test(expected = ParseException.class)
+    public void parse_standaloneInfixOperator_exceptionThrown() {
+        Parser.parse("&&");
+    }
+
+    @Test(expected = ParseException.class)
+    public void parse_standaloneSemicolon_exceptionThrown() {
+        Parser.parse(";");
+    }
+
+    @Test(expected = ParseException.class)
+    public void parse_nothingAfterInfixOperator_exceptionThrown() {
+        Parser.parse("a(b) ||");
+    }
 
-        try {
-            Parser.parse("-milestone: label:priority.high");
-            fail("'label:' can't be the input to a qualifier => empty qualifier => parse exception");
-        } catch (ParseException ignored) {}
+    @Test(expected = ParseException.class)
```

Comment:
Okay, I can change it to use ExpectedException. Maybe someone can look into AssertJ in future, as that seems to be more generic and doesn't require fields to be defined.